packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -281,6 +338,14 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
     },
     ...createListActions({...options, getKey}, fn => {
       dispatch({type: 'update', updater: fn});
-    })
+    }),
+    setFilterText(filterText: string) {
+      if (filter && data.items.length > 0) {
+        // If client side filtering and items already exist, don't sent a fetch request, just update filterText
+        dispatch({type: 'filtering', filterText});
```

Comment:
Would it make it any easier if this was just a `type: 'update'` as above and we reserved the `filtering` action for async loading?

Reply:
Yeah, I believe that would work as well. I avoided that simply because it would require adding additional `update` cases to the `loading` and `loadingMore` cases in the reducer (kinda spooky since `update` actions also happen with the other listActions above). Probably won't be too bad though, lemme try

Reply:
Ah also, are we worried about this test case: https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/data/test/useAsyncList.test.js#L647?
Seems to indicate to me that we don't want certain update actions to happen while `loading` is happening? Looking at the actions in useListData, I can see most of them being problematic if done while `loading` and `loadingMore` (except filterText heh)

Reply:
Hmm, maybe we could simplify this a bit by only allowing using `load` for filtering in `useAsyncList` rather than also a client side filter function. Client side filtering could also be implemented within `load` so I don't really see a reason to support two different ways of doing the same thing.

Reply:
Haven't tried this yet but just thinking this over: we would still need this dispatch to update the filterText right? We also wouldn't want to cancel any `loadMore` actions if filterText updated for the client side filtering case so this logic would still need exist.